src/coreclr/vm/threads.h

```diff
@@ -369,9 +369,9 @@ void SetupTLSForThread();
 // When we resume a thread at a new location, to get an exception thrown, we have to
 // pretend the exception originated elsewhere.
 EXTERN_C void ThrowControlForThread(
-#ifdef FEATURE_EH_FUNCLETS
+#if !defined(TARGET_X86)
```

Comment:
Should this use `USE_FEF` define?

Note that USE_FEF is defined for Windows x86 only: https://github.com/dotnet/runtime/blob/71cd23bbb81e94f5d865dd79544b6e1fc6bd412b/src/coreclr/vm/stackwalk.h#L39

Reply:
Hmm, I was going by the `#ifdef`s used in `Thread::ResumeUnderControl` which control the parameters passed to the method (along with `THROW_CONTROL_FOR_THREAD_FUNCTION` being set to `ThrowControlForThread` directly on x86).

The parameter is not pass on both linux-x86 and win-x86, so using `USE_FEF` as is would still leave linux-x86 broken. (On the win-x86/funclet branch I left `USE_FEF` matching linux-x86 so that would break more things; https://github.com/dotnet/runtime/pull/113576/files#diff-1970e540c30775ab1bffcd08ca1e7fa9b1a0db3f1d4fd3723cb4912154039258R39-R40)

Reply:
Would it make sense to have `USE_FEF` undefined on all x86 platforms? Perhaps I am approaching the issue from the wrong direction.

Reply:
Hmm, `USE_FEF` / `!USE_FEF` does not actually meant that the code uses / does not use `FaultingExceptionFrame`. It means that `FaultingExceptionFrame` is used differently.

> Would it make sense to have USE_FEF undefined on all x86 platforms?

I am not sure, probably not. It seems that `!USE_FEF` is yet another win-x86 quirk.